accepted/future-releases/constructor-tearoffs/faq.md

```diff
@@ -0,0 +1,262 @@
+
+Hypothetical example:
+
+```dart
+// Not a clue, mate. You tell me.
+```
+
+**Q:** Couldn't I just instantiate the `call` method using method instantiation anyway?
+
+**A:** Yes and no. You could for callable objects, and we'd even add the `.call` implicitly for you (we do that for any callable object in a function-typed context, before checking whether the types actually work). It just didn't work for *real* function values. Dart2js never implemented instantiated tear-off of the `call` method of function values because that would be equivalent to instantiating the function value itself, which wasn't a supported feature (until now), so your code would just crash. We recognized this and initially planned to disallow doing a instantiated tear-off of a function's `call` method. Instead it turned out we can just support it consistently.
+
+**Q:** Are function value instantiations canonicalized? Or equal?
+
+**A:** Since function value instantiations are never constants, they won't be canonicalized. They may be equal if the underlying instantiated functions are equal and the type arguments are the same, but they are not required to. In general, do not rely on equality of instantiated function values.
+
+## Explicit instantiation
+
+So far, you've been able to *implicitly* instantiate tear-offs. You can instantiate a *class* both implicitly *and explicitly* when doing a constructor tear-off as `List<int>.filled`. We extend that to all the other places where we currently only allow implicit instantiation. That closes a hole in the language where some type arguments could *only* be introduced by inference, but couldn't be explicitly written if you weren't satisfied with the inference result.
+
+Examples:
+
+```dart
+Type intListType = List<int>; // Explicit type literal instantiation.
+
+T id<T>(T value) => value; // Our standard generic function example.
+var idValue = id; // A function *value*.
+
+var intId = id<int>; // Explicit instantiation, saves on writing the function type.
+var intId2 = idValue<int>; // Still works!
+var intId3 = (id)<int>; // Still works!
+
+var makeList = (List.filled)<String>; // List<String> Function(int, String)
+```
+
+The last example shows the generality of the feature, because the `(List.filled)` tear-off is a generic function value, you can instantiate it. *Don't even write that*, always use `List<String>.filled` instead!
+
+In short, we allow you to use `<typeArguments>` as a *selector*, like `.name` and `[expr]`, which you can chain after an expression. Previously we only allowed type arguments to occur before an argument list (`<typeArguments>(arguments)`) or after a class name in a constructor invocation (`ClassName<typeArguments>.name(arguments)`). Now we allow it after any expression, and with the same precedence as other selectors like argument lists, `.name` and `[expr]`.
```

Comment:
`after any expression` may seem less bureaucratic than the precise information, but it could also be a source of confusion (say, if someone tries `e as T <int>`). Maybe 'Now we allow it at all locations where a `<selector>` can occur, ...'